octue/templates/template-python-fractal/app.py

```diff
@@ -1,7 +1,7 @@
 from fractal import fractal
 
 
-def run(analysis):
+def run(analysis, *args, **kwargs):
```

Comment:
Why do we need the extra args, kwargs? Is this so we can access `analysis._skip_checks`? If so, the simplest thing to do would be to expose `Analysis.skip_checks` as a protected attribute and keep the API simple.

Right now, changing this function API actually isn't a problem if there's good reason (there's about 10 apps out in the wild using this API, I need to go through and update them to the new version already) but it's not long before changing this API becomes very hard indeed.

I'm not averse in principle to having that in the templates, In fact we should keep these changes as they actually make things a lot more flexible for the future - but I don't think we should actually use it for the sake of the skip_checks parameter. 

Also good to start getting us into the mindset of being cautious of where making changes starts to cause a lot of pain and users angry with maintenance burden.

Reply:
Yeah I added them so the user-defined `run` function could accept `skip_checks` without requiring it as it only seems to be used in `DataFiles`. I think putting it on the `Analysis` object is a better idea though if it can apply to all types of analysis?